src/Persistence/MassTransit.Azure.Storage/MessageData/AzureStorageMessageDataRepository.cs

```diff
@@ -1,33 +1,28 @@
 namespace MassTransit.Azure.Storage.MessageData
 {
     using System;
+    using System.Collections.Generic;
     using System.IO;
     using System.Threading;
     using System.Threading.Tasks;
     using Context;
     using MassTransit.MessageData;
-    using Microsoft.Azure.Storage;
-    using Microsoft.Azure.Storage.Auth;
-    using Microsoft.Azure.Storage.Blob;
+    using global::Azure;
+    using global::Azure.Storage.Blobs;
+    using global::Azure.Storage.Blobs.Models;
     using Util;
 
-
     public class AzureStorageMessageDataRepository :
         IMessageDataRepository,
         IBusObserver
     {
-        readonly Uri _containerUri;
-        readonly StorageCredentials _credentials;
+        private readonly BlobContainerClient _container;
         readonly IBlobNameGenerator _nameGenerator;
 
-        public AzureStorageMessageDataRepository(Uri storageEndpoint, string containerName, StorageCredentials credentials, IBlobNameGenerator nameGenerator)
+        public AzureStorageMessageDataRepository(BlobServiceClient blobServiceClient, string containerName, IBlobNameGenerator nameGenerator)
```

Comment:
This seems like we would need to do setup work to create the client in the extension methods above.

Reply:
I decided to use the `BlobServiceClient` to encapsulate all the configuration/credential information as it provides several constructor overloads for common configuration scenarios. Reflecting on your comment I see that in that case the extension method seems no longer useful and introducing another constructor here that defaults to instantiating a `NewIdBlobNameGenerator` makes more sense, I feel.

Configuration would then be something like:
```
var blobServiceClient = new BlobServiceClient("<storage account connection string>");
IMessageDataRepository messageDataRepository = new AzureStorageMessageDataRepository(blobServiceClient, "message-data");

var busControl = Bus.Factory.CreateUsingRabbitMq(cfg =>
{
    cfg.UseMessageData(messageDataRepository);

    cfg.ReceiveEndpoint("document-service", e =>
    {
        e.Consumer<IndexDocumentConsumer>();
    });
});
```

Reply:
I guess I'd like some easy setup methods where they just pass the connection string from Config and we don't expose the underlying types to avoid future issues with dependency changes. If there is additional setup, a nested closure configuration method (similar to how most configuration works with MassTransit) could be used.

I actually thought there was a connection string only method already, but I was mistaken!

Reply:
I added overloads for `AzureStorageMessageDataRepository` to handle all the common cases. You can pass in a connection string and then choose from other options if one wants to use a different option with respect to credentials. No underlying types exposed.

I wasn't sure about introducing the nested closure configuration method since the other credential options are mutually exclusive but happy to hear your thoughts on this and I can iterate further if this isn't suitable.

If this is OK then I think the extension method above can be removed.